numpy/lib/function_base.py

```diff
@@ -3957,11 +3957,11 @@ def _quantile_is_valid(q):
     # avoid expensive reductions, relevant for arrays with < O(1000) elements
     if q.ndim == 1 and q.size < 10:
         for i in range(q.size):
-            if q[i] < 0.0 or q[i] > 1.0:
+            if not (0.0 <= q[i] <= 1.0):
                 return False
     else:
         # faster than any()
```

Comment:
```suggestion
```
If the performance tests sound alarms we can check, but I hope we can ignore that micro-opt anyway.